app/services/segment_delete_service.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+class SegmentDeleteService
+  SEGMENT_URI = URI('https://gdpr.segment.com/graphql')
+
+  class SegmentDeleteServiceError < StandardError
+    def initialize(message:, response:)
+      super(message)
+      @response = {body: response.body, code: response.code}
+    end
+
+    attr_reader :response
+
+    def error_type
+      :delete_service
+    end
+  end
+  class SegmentAuthenticationError < SegmentDeleteServiceError
+    def initialize(message: 'Cannot authenticate', response:)
+      super(message: message, response: response)
+    end
+
+    def error_type
+      :authentication
+    end
+  end
+  class SegmentUserDeletionError < SegmentDeleteServiceError
```

Comment:
I think there is no need to have two separate classes.
They could be one `SegmentError` and have some payload. And IMO we should not raise excceptions but return result, SuccessResult and ErrorResult.